### PR TITLE
ID-531 [Feat] Docs for Notifications JS API

### PR DESCRIPTION
**Pull request by squallstar**

### CODE REVIEW

File: docs/API/core/notifications.md [hunk 1]
```diff
@@ -1,5 +1,134 @@
...
+Fliplet.Navigator.Notifications.getPermission().then(function(permission) {
+  switch (permission) {
+    case 'granted':
+      // the user has granted permissions to receive notifications
+      break;
+    case 'denied':
+      // the user has denied permissions to receive notifications
+      break;
+    case 'default':
+      // no specific choice has been made yet (e.g. not requested)
```
Comment (tonytlwu) [inline docs/API/core/notifications.md:31]: In Chrome users can close the request without responding. In native apps user can kill the app before responding.

```suggestion
      // no specific choice has been made yet (e.g. not requested or user has not responded)
```

File: docs/API/core/notifications.md [hunk 2]
```diff
@@ -1,5 +1,134 @@
...
+}, function(error) {
+  // There was an error unsubscribing the user
+});
+```
+
+---
+
+### Open the App Settings
+
+Use the `Fliplet.Navigator.Notifications.openSettings()` method to open the device native settings for app where the user can manually change permissions for push notifications.
```
Comment (tonytlwu) [inline docs/API/core/notifications.md:85]:
```suggestion
Use the `Fliplet.Navigator.Notifications.openSettings()` method to open the device native settings for the app where the user can manually change permissions for push notifications.
```

File: docs/API/core/notifications.md [hunk 3]
```diff
@@ -1,5 +1,134 @@
...
+Fliplet.Hooks.on('notificationPermissionChange', function(status) {
+  // status contains "permission" and "source"
+});
+```
+
+---
+
+### Register a hook to be notified when the user subscribes for Push Notifications
+
+Use the `notificationSubscriptionChange` hook to listen for events fired when the user subscribes for push Notifications. The promise will be resolved with a `details` object containing the following keys:
```
Comment (tonytlwu) [inline docs/API/core/notifications.md:116]:
```suggestion
Use the `notificationSubscriptionChange` hook to listen for events fired when the user subscribes to push notifications. The promise will be resolved with a `details` object containing the following keys:
```

File: docs/API/core/notifications.md [hunk 4]
```diff
@@ -1,5 +1,134 @@
...
+
+- `subscriptionId`: ID of the subscription
+- `sessionId`: ID of the session (which can be used to send the notification to the device)
+- `createdAt`: DateTime (ISODATE) indicating the time the user subscribed
```
Comment (tonytlwu) [inline docs/API/core/notifications.md:120]:
```suggestion
- `createdAt`: Timestamp indicating the time the user subscribed (in ISODATE format)
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by squallstar** [comment]
@tonytlwu all suggestions applied, thanks.

**Review by tonytlwu — APPROVED** [review]

File: docs/API/core/notifications.md [hunk 5]
```diff
@@ -1,5 +1,134 @@
 # Notifications
 
+### Check if the device has permissions to receive notifications
+
+Use the `Fliplet.Navigator.Notifications.hasPermission()` method to check whether the device has permissions from the user to receive notifications (including push notifications).
```
Comment (tonytlwu) [inline docs/API/core/notifications.md:5]:
```suggestion
Use the `Fliplet.Navigator.Notifications.hasPermission()` function to check whether the device has permissions from the user to receive notifications (including push notifications).
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Pull request merged**